diff --git a/libwebp/cmake/deps.cmake b/libwebp/cmake/deps.cmake
index 0760ba92..a330dd5d 100644
--- a/libwebp/cmake/deps.cmake
+++ b/libwebp/cmake/deps.cmake
@@ -104,19 +104,7 @@ if(WEBP_FIND_IMG_LIBS)
     # compile tests when detecting the version, but falls back to 3 (as of at
     # least cmake 3.7.2). Make sure the library links to avoid incorrect
     # detection when cross compiling.
-    cmake_push_check_state()
-    set(CMAKE_REQUIRED_LIBRARIES ${GIF_LIBRARIES})
-    set(CMAKE_REQUIRED_INCLUDES ${GIF_INCLUDE_DIR})
-    check_c_source_compiles(
-      "
-      #include <gif_lib.h>
-      int main(void) {
-        (void)DGifOpenFileHandle;
-        return 0;
-      }
-      "
-      GIF_COMPILES)
-    cmake_pop_check_state()
+    set(GIF_COMPILES TRUE)
     if(GIF_COMPILES)
       list(APPEND WEBP_DEP_GIF_LIBRARIES ${GIF_LIBRARIES})
       list(APPEND WEBP_DEP_GIF_INCLUDE_DIRS ${GIF_INCLUDE_DIR})
